src/include/crc32c.h

```diff
@@ -42,10 +42,10 @@ uint32_t ceph_crc32c_zeros(uint32_t crc, unsigned length);
  */
 static inline uint32_t ceph_crc32c(uint32_t crc, unsigned char const *data, unsigned length)
 {
-#ifndef HAVE_POWER8
+#if !defined(HAVE_POWER8) && !defined(HAVE_S390X)
```

Comment:
I assume you ran a test that showed that `ceph_crc32c_zeros()` is slower than the assembly added in this PR on all-zero buffers.  It would be nice update the comment above to say
> This is faster than intel optimized assembly, but not as fast as ppc64le or s390x optimized assembly. 

for explicitness.

Reply:
It is not possible to crc zero/null buffers on the HW (lookup tables are used).

Reply:
> It is not possible to crc zero/null buffers on the HW (lookup tables are used).

Right, but `ceph_crc32c_s390x()` doesn't attempt to involve the HW in that case -- it falls back to a s390x-specific "one byte at a time" implementation in C (also based on a lookup table):

https://github.com/ceph/ceph/blob/4d5a780513d6ad4fe6140f95bf23135801a1a17b/src/common/crc32c_s390x.c#L561-L566

So the question is: which is faster on bigger buffers?  If letting Ceph's `ceph_crc32c_zeros()` do it is faster (like it is for x86), that define should be left as is.  If s390x-specific software fallback is faster, that comment should be updated to state that.

Reply:
Hi Ilya,
I do not expect the s390x software fallback is faster...

Reply:
My understanding is that ppc64le implementation was benchmarked and proved to be faster than `ceph_crc32c_zeros()`.  This is reflected in a comment:
https://github.com/ceph/ceph/blob/b7328fed0e2e9ab66b09f8134f3cebfc71214504/src/include/crc32c.h#L21-L26

I can't benchmark s390x implementation, so I will take you word for it.  One of two things needs to happen to resolve this thread:
- if you don't want `ceph_crc32c_zeros()` to be used on s390x, please update that comment with the reason
- if you are OK with `ceph_crc32c_zeros()` being used on s390x, drop changes to `src/include/crc32c.h` from the PR 

The rationale is that by default, we assume that `ceph_crc32c_zeros()` is used.  If an exception is needed for s390x similar to ppc64le, it needs to be documented.